OpenChange/MAPIStoreContactsMessage.m

```diff
@@ -1373,4 +1375,25 @@ - (void) save:(TALLOC_CTX *) memCtx
   [self updateVersions];
 }
 
+- (void) save:(TALLOC_CTX *) memCtx
+{
+  NSString *messageClass = [properties objectForKey: MAPIPropertyKey(PR_MESSAGE_CLASS_UNICODE)];
+  if (!messageClass) 
+    {
+      [self logWithFormat: @"ERROR: No message class in ContactMessage. Not saving anything"];
+      return;      
+    }
+
+  if ([messageClass isEqualToString: @"IPM.Contact"])
+    [self saveContact: memCtx]; 
+  else if ([messageClass isEqualToString: @"IPM.DistList"])
+    [self saveDistList: memCtx];
+  else 
+    {
+      [self logWithFormat: @"ERROR: Incorrect message class %@ ContactMessage. Not saving anything", messageClass];
+      return;
```

Comment:
+1 To Quique's comments, unless there's a benefit I wouldn't change the behaviour. As a matter of fact, the specifications don't limit the message classes allowed for each folder, so we may be missing something out by saving only `IPM.Contact` messages.